libbz2-rs-sys/src/decompress.rs

```diff
@@ -1009,9 +1009,40 @@ pub(crate) fn decompress(
                                 s.selector[i] = pos[0];
                             }
 
-                            t = 0;
-                            current_block = Block35;
-                            break;
+                            // try to read the coding tables in one go if there is sufficient input
+                            let bits_needed =
+                                usize::from(nGroups) * (5 + (usize::from(alphaSize) * 2 * 20));
+                            let bytes_needed = bits_needed.div_ceil(8);
+
+                            if strm.avail_in as usize >= bytes_needed {
+                                for t in 0..nGroups {
+                                    let mut curr = GET_BITS!(strm, s, 5);
+                                    for i in 0..alphaSize {
```

Comment:
```suggestion
                                for t in 0..usize::from(nGroups) {
                                    let mut curr = GET_BITS!(strm, s, 5);
                                    for i in 0..usize::from(alphaSize) {
```

This way the `s.len[usize::from(t)][usize::from(i)] = curr as u8;` below can become `s.len[t][i] = curr as u8`.